numba/tests/test_array_manipulation.py

```diff
@@ -1315,6 +1315,74 @@ def test_broadcast_arrays_incompatible_shapes_raise_valueerror(self):
                 self.assertIn("shape mismatch: objects cannot be broadcast to a single shape",
                               str(raises.exception))
 
+    def test_readonly_after_flatten(self):
+        # Reproduces Issue #8370
+        def unfold_flatten(x, y):
+            r, c = x.shape
+            a = np.broadcast_to(x, (y, r, c))
+            b = np.swapaxes(a, 0, 1)
+            cc = b.flatten()
+            d = np.reshape(cc, (-1, c))
+            d[y - 1:, :] = d[: 1 - y]
+            return d
+
+        pyfunc = unfold_flatten
+        cfunc = jit(nopython=True)(pyfunc)
+
+        # If issue #8370 is not fixed: This will fail.
+        res_nb = cfunc(np.array([[1, 1, 1], [2, 2, 2], [3, 3, 3]]), 2)
+        res_py = pyfunc(np.array([[1, 1, 1], [2, 2, 2], [3, 3, 3]]), 2)
+
+        self.assertTrue(np.array_equal(res_py, res_nb),
```

Comment:
If you use `np.testing.assert_array_equal(res_py, res_nb)` here instead of `self.assertTrue(np.array_equal(...`, then if the test fails, NumPy outputs the difference.

E.g. for a trivial example:

```
In [2]: np.testing.assert_array_equal(1, 2)

# ... snip ...

AssertionError: 
Arrays are not equal

Mismatched elements: 1 / 1 (100%)
Max absolute difference: 1
Max relative difference: 0.5
 x: array(1)
 y: array(2)
```

Reply:
sounds good, done!